### ISSUE TITLE
Supporting "unit locate" function 

### ISSUE THREAD

**Issue by TelosTelos** [Lonami/pyndustric#27]
Mlog has a function used to find blocks/ore (perhaps prioritizing ones near the currently bound unit?).  This thread is about how best to implement this in pyndustric.

One question is whether to make this a stand-along function, or a method of Unit?  Mindustry includes "unit" or "u" in its name for this function, but does not make obvious what this has to do with units, since what it actually locates is buildings or ore.  I think perhaps this automatically returns the closest match to the currently bound Unit?  If that's right, it may make sense to implement this as one or more methods of `Unit`.

Another question is what to call these methods.  In the mlog menu, this is called "unit locate", in compiled code it is "ulocate", and where you choose parameters for it, it begins with "find", which I think is my favorite.

Another question is how to specify the type(s) of target to be found.  There are 4 meta-types (ore/building/spawn/damaged), and depending on which metatype is chosen, it may accept 1-2 additional arguments.  The building version of the command accepts numerous subtypes (including core/turret/etc...) and an `enemy` boolean.  The ore version accepts an @resource (e.g. @copper) as its argument.  The other two metatypes take no additional arguments.  (I don't even know what the spawn metatype is???)  Given the differences in inputs (and also outputs) between these, I'm tempted to divide them into four separate methods, e.g. Unit.find_ore(), Unit.find_building(), Unit.find_spawn(), and Unit.find_damaged().

Yet another question is how to process the outputs of this function.  Most versions can take 4 additional variable names as input, in which it stores its outputs:  outx, outy, found, and building (except for the ore version, which omits building).  I'm not sure what the output `found` is for -- my best guess is that it's a boolean indicating whether anything was found, which probably is usually redundant with (i.e., non-zero in exactly the same circumstances as) the `building` output (except for the ore version, of course).  Is this right?  Python usually does not allow passing of variables into which to assign values, but it does allow return of multiple values, so we could support this with something like:  `position, found, building = Unit.find_building(...)`  Alternatively, if I'm right that 'found' is mostly redundant (except for the ore case), and since I'm planning to use vectorized ops for positions anyway, it may be best to simplify this to b = Unit.find_building(...), and then take care to implement the vectorized ops more or less efficiently.  (Outx and outy are somewhat redundant with using a sensor to read @x and @y attributes of buildings, which I've already set up pyndustric to do, so really the only reason to assign variables to these is to save instruction cycles, which we could do behind the scenes with compiler optimization, while giving coders the convenience of just using vectorized location operations with the building itself.  If it turns out that the found output isn't redundant, it could also be gathered silently and used in our effective compilation of bool(b), e.g., when b is used as a condition.) 

*TLDR:*  I'm leaning towards implementing 4 methods that compile to `ulocate` instructions:

found_building = Unit.find_building(type, enemy)
found_deposit = Unit.find_ore( resorce_type ) 
found_spawn??? = Unit.find_spawn... # still not sure what this is???
found_building = Unit.find_damaged()

In each case, this syntactically looks like a single value will have been returned from the function, but that value will behave object-orientedly like a Block, hence providing effective access to the other values the `ulocate` returns, and if we care about optimization, we could make the compiled code make use of those values (e.g., rather wasting instructions using a sensor to look up @x and @y coordinates of the found building later).

**Comment by bend-n** [comment]
isnt this already implemented?

https://github.com/Lonami/pyndustric/blob/03e9453835fa27218c7e72ce96b10fdea32514f8/pyndustri.pyi#L534

**Comment by Lonami** [comment]
Since the issue is fairy old and you point out to an existing locate, I'll go ahead and close.
